Source/WebCore/platform/graphics/gstreamer/GstAllocatorFastMalloc.cpp

```diff
@@ -129,39 +137,45 @@ static GstMemoryFastMalloc* gstAllocatorFastMallocMemShare(GstMemoryFastMalloc*
         memory->base.allocator, parent, memory->base.maxsize, memory->base.align,
         memory->base.offset + offset, size);
 
-    return sharedMemory;
+    return GST_MEMORY_CAST(sharedMemory);
 }
 
-static gboolean gstAllocatorFastMallocMemIsSpan(GstMemoryFastMalloc* memory, GstMemoryFastMalloc* other, gsize* offset)
+static gboolean gstAllocatorFastMallocMemIsSpan(GstMemory* baseMemory, GstMemory* otherMemory, gsize* offset)
 {
+    auto memory = GST_MEMORY_FAST_MALLOC_CAST(baseMemory);
     if (offset) {
-        auto* parent = reinterpret_cast<GstMemoryFastMalloc*>(memory->base.parent);
+        auto parent = GST_MEMORY_FAST_MALLOC_CAST(memory->base.parent);
         ASSERT(parent);
         *offset = memory->base.offset - parent->base.offset;
     }
 
-    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // GLib port
-    return memory->data + memory->base.offset + memory->base.size == other->data + other->base.offset;
-    WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
+    auto thisSpan = memory->data.subspan(memory->base.offset, memory->base.size);
+
+    auto other = GST_MEMORY_FAST_MALLOC_CAST(otherMemory);
+    auto otherSpan = other->data.subspan(other->base.offset);
+    return thisSpan.data() == otherSpan.data();
```

Comment:
This is not correct. The original formula was:
```c
return memory->data + memory->base.offset + memory->base.size == other->data + other->base.offset;
```
However, this is equivalent to:
```c
return memory->data + memory->base.offset == other->data + other->base.offset;
```

This is an insidious bug because WebKit will run fine in your computer, but you will have lost zero-copy on elements handling chunked data like demuxers.

Instead you can do:

```cpp
auto visibleSpan1 = memory1->data.subspan(memory1->base.offset, memory1->base.size);
auto visibleSpan2 = memory2->data.subspan(memory2->base.offset);
return visibleSpan1.end() == visibleSpan2.begin();
```